containers/product/index.js

```diff
@@ -18,12 +19,47 @@ export default function Product ({ slug }) {
 
   const { productBySlug } = data
 
+  const hasVariants = () => productBySlug.variants.nodes.length > 0
+  const defaultVariantId = () =>
+    hasVariants()
+      ? productBySlug.variants.nodes[0].id
+      : productBySlug.masterVariant.id
+
   return (
     <div>
       <h1>{productBySlug.name}</h1>
-      <button onClick={() => {
-        addToCart({ variables: { variantId: productBySlug.masterVariant.id, quantity: 1 } });
-      }}>Add to cart</button>
+      <Formik
+        initialValues={{
+          quantity: 1,
+          variantId: defaultVariantId()
+        }}
+        onSubmit={async values => {
```

Comment:
I don't think we need the `async` here, since we are not using `await` inside the function. Also, could we refactor this function out, in order to declutter the `return`/`render` of the component?

Reply:
👍